USFMToolsSharp.Renderers.USX/USXRenderer.cs

```diff
@@ -0,0 +1,184 @@
+                    
+                    output.Append($"<book style=\"{input.Identifier}\" code=\"{bookCode}\">");
+                    output.Append(bibleVersion);
+                    output.AppendLine("</book>");
+                    break;
+                
+                case USFMMarker usfmMarker:
+                    output.AppendLine($"<para style=\"{usfmMarker.Identifier}\">{usfmMarker.Version}</para>");
+                    break;
+                
+                case IDEMarker ideMarker:
+                    output.AppendLine($"<para style=\"{ideMarker.Identifier}\">{ideMarker.Encoding}</para>");
+                    break;
+                    
+                case HMarker hMarker:
+                    output.AppendLine($"<para style=\"{hMarker.Identifier}\">{hMarker.HeaderText}</para>");
+                    break;
+                
+                case TOC1Marker toc1Marker:
+                    output.AppendLine($"<para style=\"{toc1Marker.Identifier}\">{toc1Marker.LongTableOfContentsText}</para>");
+                    break;
+
+                case TOC2Marker toc2Marker:
+                    output.AppendLine($"<para style=\"{toc2Marker.Identifier}\">{toc2Marker.ShortTableOfContentsText}</para>");
+                    break;
+                
+                case TOC3Marker toc3Marker:
+                    output.AppendLine($"<para style=\"{toc3Marker.Identifier}\">{toc3Marker.BookAbbreviation}</para>");
+                    break;
+                
+                case MTMarker mtMarker:
+                    output.AppendLine($"<para style=\"{mtMarker.Identifier}\">{mtMarker.Title}</para>");
+                    break;
+                
+                case SMarker sMarker:
+                    output.AppendLine($"<para style=\"{sMarker.Identifier}{sMarker.Weight}\">{sMarker.Text}</para>");
+                    break;
+                
+                case PMarker pMarker:
+                    output.AppendLine($"<para style=\"{pMarker.Identifier}\">");
+
+                    foreach (Marker marker in input.Contents)
+                    {
+                        output.Append(RenderMarker(marker));
+                    }
+
+                    output.AppendLine("</para>");
+                    break;
+                
+                case CMarker cMarker:
+
+                    // If the next chapter is found, mark the end of the current chapter
+                    if (CurrentChapter < cMarker.Number)
+                    {
+                        output.AppendLine($"<chapter eid=\"{CurrentBookCode} {CurrentChapter}\" />");
+                    }
+                    
+                    CurrentChapter = cMarker.Number;
+                    output.AppendLine($"<chapter style=\"{cMarker.Identifier}\" " +
+                                      $"number=\"{cMarker.Number}\" " +
+                                      $"sid=\"{CurrentBookCode} {cMarker.Number}\" />");
```

Comment:
We need to make USX 3.0 support configurable because BTTWriter doesn't support it.  I would love to have USX 3.0 support but if it is too much work let's just skip it. If you can add USX 3.0 support that would be nice.

Reply:
I put USXVersion inside the USXConfig file, would that make it configurable, then? I'm not sure what you meant by having a USX 3.0 support.

Reply:
Oh, did you mean that in USX version lower than 3.0 doesn't have "sid" and "eid" fields?